### PR TITLE
fix: improve fix for pz 851

### PR DESCRIPTION
**Pull request by edgarvonk**
Improve fix for PZ-851 - it should not be possible to add documents to finalized tasks - by adding a new 'taak - toevoegenDocument' OPA policy.

Solves PZ-1444

### CODE REVIEW

File: src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java [hunk 1]
```diff
@@ -307,6 +307,7 @@ public RESTEnkelvoudigInformatieobject createEnkelvoudigInformatieobject(
                         Response.Status.CONFLICT
                 );
             }
+            assertPolicy(policyService.readTaakRechten(task, zaak).toevoegenDocument());
```
Comment (bas-info-nl) [inline src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:310]: Multi-lingual variables 🙀

File: src/main/java/net/atos/zac/policy/output/TaakRechten.java [hunk 2]
```diff
@@ -9,16 +9,23 @@
...
+) implements DummyInterface {
 
     @JsonbCreator
     public TaakRechten(
             @JsonbProperty("lezen") final boolean lezen,
             @JsonbProperty("wijzigen") final boolean wijzigen,
-            @JsonbProperty("toekennen") final boolean toekennen) {
+            @JsonbProperty("toekennen") final boolean toekennen,
+            @JsonbProperty("toevoegen_document") final boolean toevoegenDocument
+    ) {
```
Comment (bas-info-nl) [inline src/main/java/net/atos/zac/policy/output/TaakRechten.java:25]: Lenient creation? Use `Boolean` parameters and primitive `boolean` properties?

File: src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java (hunk 1 above)
Comment (edgarvonk) [reply src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:310]: yep. ga ik niks aan doen voorlopig hoor. ;-)

File: src/main/java/net/atos/zac/policy/input/TaakData.java [hunk 3]
```diff
@@ -5,7 +5,15 @@
...
 package net.atos.zac.policy.input;
 
+import jakarta.json.bind.annotation.JsonbProperty;
+
 public class TaakData {
 
+    public boolean open;
+
+    @JsonbProperty("zaak_open")
+    public Boolean zaakOpen;
```
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/java/net/atos/zac/policy/input/TaakData.java]: Good one. Converted them to `boolean`. I copied this code from another class where Boolean and boolean were used. Fixed that one too.

**Review by bas-info-nl — APPROVED** [review]

File: src/main/java/net/atos/zac/policy/output/TaakRechten.java (hunk 2 above)
Comment (bas-info-nl) [reply src/main/java/net/atos/zac/policy/output/TaakRechten.java:25]: This was in respect to [Postel's Law](https://deviq.com/laws/postels-law), but reconsidering this in respect to internal APIs we should be strict. Nothing to do here 😅

**Pull request merged**